From bbc19042f7c15e9a9867e951a302ff46225fad31 Mon Sep 17 00:00:00 2001
From: Mario Schuknecht <mario.schuknecht@dresearch-fe.de>
Date: Fri, 9 Mar 2018 12:31:24 +0100
Subject: [PATCH] tty: imx: Disable DMA HYP-18742, HYP-14147

Disable imx UART DMA again, because there are too many problems.

Signed-off-by: Mario Schuknecht <mario.schuknecht@dresearch-fe.de>
---
 drivers/tty/serial/imx.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/tty/serial/imx.c b/drivers/tty/serial/imx.c
index 8bae46c8119d..8f427a1ca0e9 100644
--- a/drivers/tty/serial/imx.c
+++ b/drivers/tty/serial/imx.c
@@ -1302,10 +1302,13 @@ static int imx_startup(struct uart_port *port)
 
 	writel(temp & ~UCR4_DREN, sport->port.membase + UCR4);
 
+#if 0
+Disable UART DMA, because too many problems. HYP-18742, HYP-14147, HYP-14546, HYP-15101
 	/* Can we enable the DMA support? */
 	if (is_imx6q_uart(sport) && !uart_console(port) &&
 	    !sport->dma_is_inited)
 		imx_uart_dma_init(sport);
+#endif
 
 	if (sport->dma_is_inited)
 		INIT_WORK(&sport->tsk_dma_tx, dma_tx_work);
-- 
2.13.5

